### PR TITLE
Enable Nullable

### PR DESCRIPTION
**Pull request by maxkoshevoi**
**Related issues**
- Fixes: #79

**What was done**
- Enabled nullable feature for all non-test projects
- Converted some switches to switch expressions

**Notes**
- Tried to didn't alter any behavior
- Didn't enable nullable for tests
- There are couple of changes like this: 
image
They are needed because of an old SDK.  `IsNullOrEmpty` has an `NotNullWhen(false)` attribute in a latter SDK, so after update to .Net 6, they won't be needed.
- I'm not sure about setting `Calendar.Name` as nullable since it doesn't seem like calendar can exist without it. Calendars without name don't show up in `Google calendar` in Android, and in iOS `Title` is non-nullable, so I added an exception when it is: 
image
- I needed to add `#nullable enable` to all files for VS to work for some reason. Everything should work with just `Directory.Build.props`. Maybe it's because an old SDK.

**Tests**
- UWP tests succeeded:
image
- Android tests succeeded:
image
- Don't have access to iOS device (or Mac) to perform iOS tests

### CODE REVIEW

File: Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs [hunk 1]
```diff
@@ -90,20 +79,18 @@ public async Task<Calendar> GetCalendarByIdAsync(string externalId)
...
-                throw new ArgumentException("Specified calendar not found on device");
-            }
+            var deviceCalendar = _eventStore.GetCalendar(calendar.ExternalID)
+                ?? throw new ArgumentException("Specified calendar not found on device");
 
             var query = _eventStore.PredicateForEvents(start.ToNSDate(), end.ToNSDate(), new EKCalendar[] { deviceCalendar });
             var events = await Task.Run(() =>
             {
                 var iosEvents = _eventStore.EventsMatching(query);
-                return iosEvents == null ? new List<CalendarEvent>() : iosEvents.Select(e => e.ToCalendarEvent()).ToList();
```
Comment (maxkoshevoi) [inline Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs:106]: This change is not needed, but `EventsMatching` cannot return `null` according to it's attributes.
If this is a mistake, I'll revert it.

File: Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs [hunk 2]
```diff
@@ -90,20 +79,18 @@ public async Task<Calendar> GetCalendarByIdAsync(string externalId)
...
+                return iosEvents.Select(e => e.ToCalendarEvent()).ToList();
```
Comment (maxkoshevoi) [inline Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs:93]: This change is not needed, but `EventsMatching` cannot return `null` according to it's attributes.
If this is a mistake, I'll revert it.

**Comment by TheAlmightyBob** [comment]
Thanks! It'll take me some time to get to this, but I appreciate what you've done here and the helpful description.

Re:
> VS doesn't see Android or iOS tests

Yeah, the Android/iOS tests have to be run on a device/simulator. You just run those projects like you would an app and tap the "run tests" button (there's some way to have them autorun and log results but it's been too long since I've done that..).

File: Calendars/Calendars.Plugin.Android/DateConversions.cs [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
+
 namespace Plugin.Calendars
 {
     /// <summary>
     /// Date conversion helpers.
     /// </summary>
     internal static class DateConversions
     {
-        private readonly static DateTime _reference = new DateTime(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
+        private readonly static DateTime _reference = new(1970, 1, 1, 0, 0, 0, DateTimeKind.Utc);
```
Comment (TheAlmightyBob) [inline Calendars/Calendars.Plugin.Android/DateConversions.cs:12]: Woah. I didn't know about this...

Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.Android/DateConversions.cs:12]: This change doesn't alter any behavior (or even resulting IL code). Are you against C# 9 `new` syntax, or is there some other issue?

Comment (TheAlmightyBob) [reply Calendars/Calendars.Plugin.Android/DateConversions.cs:12]: No issue at all! I'm impressed. I didn't know about this syntax.

Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.Android/DateConversions.cs:12]: I mean, this syntax was designed exactly for this: not having to duplicate type when assigning default value to properly or field

Comment (TheAlmightyBob) [reply Calendars/Calendars.Plugin.Android/DateConversions.cs:12]: Yep, makes perfect sense.

Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.Android/DateConversions.cs:12]: Sorry, misread your first comment 😄
Yeah, that's a cool one. Basically you can now write `new()` (without type) if type can be inferred by the compiler. So, for example `new List<Point> { new(1, 2) }` is also valid, or `Point p = new(1, 2)`

File: Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs [hunk 4]
```diff
@@ -162,16 +147,18 @@ public async Task AddOrUpdateCalendarAsync(Calendar calendar)
...
             }
             else
             {
                 deviceCalendar.Title = calendar.Name;
 
-                if (!string.IsNullOrEmpty(calendar.Color))
+                if (calendar.Color != null && !string.IsNullOrEmpty(calendar.Color))
                 {
-                    deviceCalendar.CGColor = ColorConversion.ToCGColor(calendar.Color);
+                    deviceCalendar.CGColor = ColorConversion.ToCGColor(calendar.Color) ?? throw new InvalidOperationException("Invalid color");
```
Comment (TheAlmightyBob) [inline Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs:161]: This seems like it should be an `ArgumentException`, not `InvalidOperationException`. The color was passed-in via the `calendar` argument.

**Review by TheAlmightyBob — CHANGES_REQUESTED** [review]
I pulled this up in VS4Mac and ran the iOS/Android tests; things generally look good! Just a few small questions/requests.

File: Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs (hunk 2 above)
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs:93]: `var` is nullable by default, but compiler constantly checks code flow. So if you assign non-null value to var, there will be no warnings when using it, but there'll be a warning if you assign null to var and then try to use it.

File: Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs [hunk 5]
```diff
@@ -142,18 +129,16 @@ public async Task<CalendarEvent> GetEventByIdAsync(string externalId)
         /// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
         public async Task AddOrUpdateCalendarAsync(Calendar calendar)
         {
+            _ = calendar.Name ?? throw new ArgumentNullException("Name cannot be null");
```
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs:132]: This is shorter. Other than that - no reason. Do you prefer `if`?

File: Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs [hunk 6]
```diff
@@ -19,9 +21,9 @@ public class CalendarsImplementation : ICalendars
     {
         #region Constants
 
-        private static readonly Android.Net.Uri _calendarsUri = CalendarContract.Calendars.ContentUri;
-        private static readonly Android.Net.Uri _eventsUri = CalendarContract.Events.ContentUri;
-        private static readonly Android.Net.Uri _remindersUri = CalendarContract.Reminders.ContentUri;
+        private static readonly Android.Net.Uri? _calendarsUri = CalendarContract.Calendars.ContentUri;
+        private static readonly Android.Net.Uri? _eventsUri = CalendarContract.Events.ContentUri;
+        private static readonly Android.Net.Uri? _remindersUri = CalendarContract.Reminders.ContentUri;
```
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs:26]: No, Android libraries have been updated to support nullable. But it seems like they overdid it with `?` a bit. Maybe nullable situation it more accurate in `.net6-android`, but I'm not sure.

File: Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs [hunk 7]
```diff
@@ -92,25 +91,25 @@ public Task<IList<Calendar>> GetCalendarsAsync()
...
         /// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
-        public Task<Calendar> GetCalendarByIdAsync(string externalId)
+        public Task<Calendar?> GetCalendarByIdAsync(string? externalId)
         {
             long calendarId = -1;
 
             if (!long.TryParse(externalId, out calendarId))
             {
-                return null;
+                return Task.FromResult<Calendar?>(null);
```
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs:100]: It was caught be nullable feature. Cause in the original form I method would need to return `Task<Calendar?>?`

File: Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs [hunk 8]
```diff
@@ -132,7 +131,8 @@ public async Task<IList<CalendarEvent>> GetEventsAsync(Calendar calendar, DateTi
                 throw new ArgumentException("Specified calendar not found on device");
             }
 
-            var eventsUriBuilder = CalendarContract.Instances.ContentUri.BuildUpon();
+            var eventsUriBuilder = CalendarContract.Instances.ContentUri?.BuildUpon()
+                    ?? throw new NullReferenceException("CalendarContract.Instances.ContentUri");
```
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs:135]: `NullReferenceException` wouldn't have any message with `!.`. 
I think `BuildUpon` is null only if `CalendarContract.Instances.ContentUri` is null. Wil need to double check that.

File: Calendars/Calendars.Plugin.UWP/CalendarsImplementation.cs [hunk 9]
```diff
@@ -411,8 +410,12 @@ public async Task<bool> DeleteEventAsync(Calendar calendar, CalendarEvent calend
 
 #region Private Methods
 
+        [MemberNotNull(nameof(_apptStore), nameof(_localApptStore))]
         private async Task EnsureInitializedAsync()
         {
+            // If EnsureInitializedAsync is not awaited, they won't be initialized. It's fine.
+#pragma warning disable CS8774 // Member must have a non-null value when exiting.
```
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.UWP/CalendarsImplementation.cs:417]: `async` initialization works, `sync` - isn't. As long as `EnsureInitializedAsync` is always awaited, everything should work fine.

File: Calendars/Calendars.Plugin/CrossCalendars.cs [hunk 10]
```diff
@@ -1,14 +1,17 @@
...
 using System;
+using System.Threading;
+
+#nullable enable
 
 namespace Plugin.Calendars
 {
   /// <summary>
-  /// Cross platform Calendars implemenations
+  /// Cross platform Calendars implementations
```
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin/CrossCalendars.cs:10]: No problem. There's also `writeable` -> `writable` misspelling everywhere. Didn't correct it since it would change public API.

File: Calendars/Calendars.Plugin.Android/CalendarEventReminderExtensions.cs [hunk 11]
```diff
@@ -2,11 +2,13 @@
 using Android.Provider;
 using Plugin.Calendars.Abstractions;
 
+#nullable enable
```
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.Android/CalendarEventReminderExtensions.cs:5]: I'm getting a lot of those if `#nullable enable` is removed

image

(everything still builds though)

File: Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs [hunk 12]
```diff
@@ -142,18 +129,16 @@ public async Task<CalendarEvent> GetEventByIdAsync(string externalId)
...
+
             await RequestCalendarAccess().ConfigureAwait(false);
 
-            EKCalendar deviceCalendar = null;
+            EKCalendar? deviceCalendar = null;
 
-            if (!string.IsNullOrEmpty(calendar.ExternalID))
+            if (calendar.ExternalID != null && !string.IsNullOrEmpty(calendar.ExternalID))
```
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs:139]: Added `TODO: Remove redundant null check after migrating to .net6` comment everywhere.

File: Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs (hunk 8 above)
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs]: But yeah, maybe `!.` is cleaner in this case. Removed NRE in favor of `!.`

File: Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs (hunk 4 above)
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs]: Fixed

File: Calendars/Calendars.Plugin/CrossCalendars.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (TheAlmightyBob) [reply Calendars/Calendars.Plugin/CrossCalendars.cs:10]: Well, that one I can kind of blame on the fact that `writeable` is also an "accepted" spelling that wouldn't be caught by spellcheck... but you're right that "writable" is the more standard spelling.

However, I don't see it appearing in the public API...? All the instances I see are `internal` or `private` (it's not a big deal, just curious if I'm missing something)

File: Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (TheAlmightyBob) [reply Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs:132]: I would prefer `if` in this case. I can understand this pattern in the other cases where the actual purpose is to perform the assignment, but in this particular instance I think the fake assignment is a tad bit too clever and hurts readability.

File: Calendars/Calendars.Plugin.Android/CalendarEventReminderExtensions.cs (hunk 11 above)
[the remark this answers is not in the document]
Comment (TheAlmightyBob) [reply Calendars/Calendars.Plugin.Android/CalendarEventReminderExtensions.cs:5]: That is very curious. What version of VS are you using?

Perhaps VS4Mac just isn't as strict about that warning... though I thought for things like this they both used the same code thanks to Roslyn...

For _science_ I'd be somewhat curious if putting the config in those project files (rather than just relying on directory.build.props) would make a difference... but I s'pose `#nullable enable` isn't the end of the world...

File: Calendars/Calendars.Plugin.UWP/CalendarsImplementation.cs (hunk 9 above)
[the remark this answers is not in the document]
Comment (TheAlmightyBob) [reply Calendars/Calendars.Plugin.UWP/CalendarsImplementation.cs:421]: Yeah I meant that I was surprised the compiler didn't detect that and enforce `MemberNotNull` asynchronously... though I guess that could be a hard problem...

File: Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (TheAlmightyBob) [reply Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs:26]: Curious. I looked through the Android libraries in the assembly browser and it really looked like _everything_ had `?`...

File: Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs (hunk 8 above)
[the remark this answers is not in the document]
Comment (TheAlmightyBob) [reply Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs]: > I think BuildUpon is null only if CalendarContract.Instances.ContentUri is null

You may be right in practice, but judging from the way the API is currently annotated it claims that `BuildUpon` can itself return `null`, separately from whether `ContentUri` is null... as noted in the other comment though, it seems like _everything_ is annotated as possibly returning `null`, so that could be a lie...

File: Tests/Calendars.Plugin.UWP.Tests/Calendars.Plugin.UWP.Tests.csproj [hunk 13]
```diff
@@ -142,16 +142,16 @@
...
-      <Version>6.2.9</Version>
+      <Version>6.2.12</Version>
     </PackageReference>
     <PackageReference Include="MSTest.TestAdapter">
-      <Version>2.0.0</Version>
+      <Version>2.2.5</Version>
     </PackageReference>
     <PackageReference Include="MSTest.TestFramework">
-      <Version>2.0.0</Version>
+      <Version>2.2.5</Version>
```
Comment (TheAlmightyBob) [inline Tests/Calendars.Plugin.UWP.Tests/Calendars.Plugin.UWP.Tests.csproj:154]: I'm not opposed to these changes, but I'd _generally_ prefer them to be a separate PR unless they're directly related. (and the "cause why not" suggests that they're not related)

File: Calendars/Calendars.Plugin/CrossCalendars.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin/CrossCalendars.cs:10]: You are right. Missed that class is internal here

image

If you don't mind, I can also change `writeable` to `writable`

File: Calendars/Calendars.Plugin.Android/CalendarEventReminderExtensions.cs (hunk 11 above)
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.Android/CalendarEventReminderExtensions.cs:5]: I'm using latest VS 2019 Enterprise (16.10.4). Interesting thing is that I don't have this probe in my Xamarin project that targets `.netstandart2.1`.

File: Calendars/Calendars.Plugin.UWP/CalendarsImplementation.cs (hunk 9 above)
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.UWP/CalendarsImplementation.cs:421]: That's a good point. ~~I guess that's because this attribute is designed to be used by constructors: https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/attributes/nullable-analysis#constructor-helper-methods-membernotnull-and-membernotnullwhen~~
Nope this attribute is intended for exactly this purpose: https://github.com/dotnet/runtime/issues/31877

Having `MemberNotNull` work with async methods would be a great improvement, I agree.

[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.UWP/CalendarsImplementation.cs:421]: Created an issue: https://github.com/dotnet/runtime/issues/56356

File: Tests/Calendars.Plugin.UWP.Tests/Calendars.Plugin.UWP.Tests.csproj (hunk 13 above)
Comment (maxkoshevoi) [reply Tests/Calendars.Plugin.UWP.Tests/Calendars.Plugin.UWP.Tests.csproj:154]: Sure thing. Reverted the change

File: Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs (hunk 5 above)
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs]: Refactored null check to use `if`.

File: Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs (hunk 6 above)
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.Android/CalendarsImplementation.cs:26]: Seems like you are right. Every public member of `Uri` class is nullable. But it's not a default thing. Those attributes were explicitly  added there: https://github.com/xamarin/xamarin-android/pull/4227/files

**Review by TheAlmightyBob — CHANGES_REQUESTED** [review]

File: Calendars/Calendars.Plugin/CrossCalendars.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (TheAlmightyBob) [reply Calendars/Calendars.Plugin/CrossCalendars.cs:10]: If it's no trouble for you, I would appreciate the correction there. Certainly not required for this MR though. :)

File: Calendars/Calendars.Plugin.Android/CalendarEventReminderExtensions.cs (hunk 11 above)
[the remark this answers is not in the document]
Comment (TheAlmightyBob) [reply Calendars/Calendars.Plugin.Android/CalendarEventReminderExtensions.cs:5]: Alright, thanks for checking.

File: Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs [hunk 14]
```diff
@@ -90,20 +79,18 @@ public async Task<Calendar> GetCalendarByIdAsync(string externalId)
         /// <exception cref="Plugin.Calendars.Abstractions.PlatformException">Unexpected platform-specific error</exception>
         public async Task<IList<CalendarEvent>> GetEventsAsync(Calendar calendar, DateTime start, DateTime end)
         {
-            await RequestCalendarAccess().ConfigureAwait(false);
+            _ = calendar.ExternalID ?? throw new ArgumentNullException("ExternalID cannot be null");
```
[the remark this answers is not in the document]
Comment (TheAlmightyBob) [reply Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs:82]: (sorry I wrote this comment earlier with all the others but apparently forgot to click "send")

[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin.iOSUnified/CalendarsImplementation.cs]: Forgot I did that in more than one place too. Thanks, fixed it =)

File: Calendars/Calendars.Plugin/CrossCalendars.cs (hunk 10 above)
[the remark this answers is not in the document]
Comment (maxkoshevoi) [reply Calendars/Calendars.Plugin/CrossCalendars.cs:10]: Done

**Review by TheAlmightyBob — APPROVED** [review]

**Comment by TheAlmightyBob** [comment]
Looks good, thanks! I'll get this merged soon.

**Comment by TheAlmightyBob** [comment]
Argh, the CI build failed, so it could be a while before a new release. I doubt the failure has anything to do with this change, probably just it's been so long since the last release that something else needs to be updated there...

**Comment by maxkoshevoi** [comment]
@TheAlmightyBob Could you please take a look at https://github.com/TheAlmightyBob/Calendars/issues/79#issuecomment-887311824

**Comment by maxkoshevoi** [comment]
> Argh, the CI build failed, so it could be a while before a new release.

Hi, any news about that? Maybe I could help fix CI somehow?

**Comment by TheAlmightyBob** [comment]
I appreciate the offer to help, but unless you're familiar with https://cakebuild.net/, it may be a waste of your time. It looks like they made breaking changes since my last release, and the CI script pulls the latest version of Cake rather than specifying a version. So _hopefully_ it's not too big a deal to either adapt to the breaking changes or lock to the previously-working version.

(It would be nice if the CI was hooked up to MRs so this would have been caught earlier.... I looked into that years ago, but don't remember what the problem was... perhaps a limitation of Appveyor...)

(delay was due to some emergencies at home, only just starting to look at this now)

**Comment by TheAlmightyBob** [comment]
Hm, okay, I fixed the original CI version issue, but then things got more interesting.

https://ci.appveyor.com/project/TheAlmightyBob/calendars/builds/40387992 failed with lots of errors like this:
> error CS0453: The type '[various types here]' must be a non-nullable value type in order to use it as parameter 'T' in the generic type or method 'Nullable<T>'

That build was using VS2017, so I tried switching to VS2019, which instead gives a bunch of _these_ errors:
> error CS1069: The type name 'AppointmentStore' could not be found in the namespace 'Windows.ApplicationModel.Appointments'. This type has been forwarded to assembly 'Windows, Version=255.255.255.255, Culture=neutral, PublicKeyToken=null, ContentType=WindowsRuntime' Consider adding a reference to that assembly.

&

> error CS0731: The type forwarder for type 'Windows.ApplicationModel.Appointments.AppointmentCalendar' in assembly 'Windows' causes a cycle 

But I trust you didn't see anything like this when building in VS....

**Comment by maxkoshevoi** [comment]
> But I trust you didn't see anything like this when building in VS.

Very strange. I used VS 2019 and didn't see anything (I was able to run all the test, so build must've passed). I'll recheck just to be sure.

**Comment by maxkoshevoi** [comment]
Maybe the issue is that I used different Windows SDK (VS automatically updated your project to this one since I had it installed):

image

I don't seem to be able to install `10.0.10586.0` though. Here're all versions that I have in VS installer (sorry for Russian):

image

**Comment by TheAlmightyBob** [comment]
Interesting, thanks for sharing. It might not hurt to update the target version... I suppose 10586 is rather old now...

I'm a little surprised that it also changed the minimum version, but again probably not a big deal...

**Comment by maxkoshevoi** [comment]
> I'm a little surprised that it also changed the minimum version, but again probably not a big deal

Well, it doesn't do it automatically, I still have to agree to it:

image

**Comment by TheAlmightyBob** [comment]
But that just says _target_ version, not _minimum_ version. Those are separate things in the project file.

**Comment by maxkoshevoi** [comment]
> But that just says _target_ version, not _minimum_ version.

Didn't even notice that it also changes min version. But it is not mandatory. It still builds fine if I revert min version to it's original value. Only changing target version should be enough.

**Comment by TheAlmightyBob** [comment]
Alright, updated the target version, and it _claims_ to have succeeded now.... 🤞
